### DISCUSSION TITLE
[Q&A - CHAPTER 2] 동작 파라미터화 코드 전달하기

### DISCUSSION THREAD

**Discussion by Irisation23** [Study-2-Modern-Java-In-Action/Study-2-Modern-Java-in-Action#20] [CHAPTER 2 동작 파라미터화 코드 전달하기 - Q&A]
- 댓글 남겨주세요~

**Comment by JoisFe** [comment]
chapter 1 에서 `자바 8에서는 메서드(우리 코드)를 다른 메서드의 인수로 넘겨주는 기능을 제공한다. 45pg`, `자바 8 설계자들은 이급 시민을 일급 시민으로 바꿀 수 있는 기능을 추가했다. 49pg`
- 해당 내용이 엄밀하게 맞는 것 일까요?
- 람다, 메서드 참조를 넘기면서 그렇게 보이는 것이지 엄밀하게 익명 클래스의 인스턴스로 알 고 있는데 메서드 자체를 인수로 넘겨주는 것이 아니므로 일급이 아니지 않은지

chapter 2 에서 `나중에 실행될 메서드의 인수로 코드 블록을 전달할 수 있다. 결과적으로 코드 블록에 따라 메서드의 동작이 파라미터화 된다`
와 같이 `메서드의 동작이 파라미터화 된다` 라는 말이 더 정확한 표현이 맞지 않은지? 궁금합니다.

**Comment by Irisation23** [comment]
### Q. `() ->` 는 자바에만 있을까요?

**자바스크립트**의 `() => ` 도 있죠
**자바**의 `() -> `와 **자바스크립트**의 `() => `는 어떤 차이가 있을까요??
